rust/vectordb/src/table.rs

```diff
@@ -665,24 +676,27 @@ impl NativeTable {
         options: CompactionOptions,
         remap_options: Option<Arc<dyn IndexRemapperOptions>>,
     ) -> Result<CompactionMetrics> {
-        let mut dataset = self.clone_inner_dataset();
-        let metrics = compact_files(&mut dataset, options, remap_options).await?;
-        self.reset_dataset(dataset);
+        let mut dataset_mut = self.dataset.get_mut().await?;
+        let metrics = compact_files(&mut dataset_mut, options, remap_options).await?;
         Ok(metrics)
     }
 
-    pub fn count_fragments(&self) -> usize {
-        self.dataset.lock().expect("lock poison").count_fragments()
+    // TODO: why are these individual methods and not some single "get_stats" method?
```

Comment:
+1, they also don't exist on the other APIs so they will need to be reconciled at some point.  Plus, they are only part of `NativeTable` and not `Table`.  I think these kind of get added piecemeal / ad-hoc as needed for internal use cases maybe?